cibuildwheel/macos.py

```diff
@@ -117,64 +180,8 @@ def build(project_dir, output_dir, test_command, before_test, test_requires, tes
-        env['PATH'] = os.pathsep.join([
-            SYMLINKS_DIR,
-            installation_bin_path,
-            env['PATH'],
-        ])
-
-        # Fix issue with site.py setting the wrong `sys.prefix`, `sys.exec_prefix`,
-        # `sys.path`, ... for PyPy: https://foss.heptapod.net/pypy/pypy/issues/3175
-        # Also fix an issue with the shebang of installed scripts inside the
-        # testing virtualenv- see https://github.com/theacodes/nox/issues/44 and
-        # https://github.com/pypa/virtualenv/issues/620
-        # Also see https://github.com/python/cpython/pull/9516
-        env.pop('__PYVENV_LAUNCHER__', None)
-        env = environment.as_dictionary(prev_environment=env)
-
-        # check what version we're on
-        call(['which', 'python'], env=env)
-        call(['python', '--version'], env=env)
-        which_python = subprocess.check_output(['which', 'python'], env=env, universal_newlines=True).strip()
-        if which_python != '/tmp/cibw_bin/python':
-            print("cibuildwheel: python available on PATH doesn't match our installed instance. If you have modified PATH, ensure that you don't overwrite cibuildwheel's entry or insert python above it.", file=sys.stderr)
-            exit(1)
-
-        dependency_constraint_flags = []
-        if dependency_constraints:
-            dependency_constraint_flags = [
-                '-c', dependency_constraints.get_for_python_version(config.version)
-            ]
-
-        # install pip & wheel
-        call(['python', get_pip_script] + dependency_constraint_flags, env=env, cwd="/tmp")
-        assert os.path.exists(os.path.join(installation_bin_path, 'pip'))
-        call(['which', 'pip'], env=env)
-        call(['pip', '--version'], env=env)
-        which_pip = subprocess.check_output(['which', 'pip'], env=env, universal_newlines=True).strip()
-        if which_pip != '/tmp/cibw_bin/pip':
-            print("cibuildwheel: pip available on PATH doesn't match our installed instance. If you have modified PATH, ensure that you don't overwrite cibuildwheel's entry or insert pip above it.", file=sys.stderr)
-            exit(1)
-        call(['pip', 'install', '--upgrade', 'setuptools', 'wheel', 'delocate'] + dependency_constraint_flags, env=env)
-
-        # setup target platform, only required for python 3.5
-        if config.version == '3.5':
-            if '_PYTHON_HOST_PLATFORM' not in env:
-                # cross-compilation platform override
-                env['_PYTHON_HOST_PLATFORM'] = 'macosx-10.9-x86_64'
-            if 'ARCHFLAGS' not in env:
-                # https://github.com/python/cpython/blob/a5ed2fe0eedefa1649aa93ee74a0bafc8e628a10/Lib/_osx_support.py#L260
-                env['ARCHFLAGS'] = '-arch x86_64'
-            if 'MACOSX_DEPLOYMENT_TARGET' not in env:
-                env['MACOSX_DEPLOYMENT_TARGET'] = '10.9'
+        env, dependency_constraint_flags = setup_python(config, dependency_constraints, environment)
```

Comment:
I don't like the multiple return here, the `env` makes a lot of sense, but the dependency_constraint_flags seems too trivial. Could that go elsewhere? I think a method `DependencyConstraints.pip_flags(python_version)` would neaten this up.

Reply:
I move `dependecy_constrains_flags` outside `python_setup` function.